packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -625,37 +640,59 @@ class _RootBuilder extends dart_ast_visitor.RecursiveAstVisitor<Object?> {
         );
       }
     } else {
-      _currentClass = Class(name: node.name.name, fields: <Field>[]);
+      _currentClass = Class(name: node.name.name, fields: <NamedType>[]);
     }
 
     node.visitChildren(this);
     return null;
   }
 
+  NamedType formalParameterToField(dart_ast.FormalParameter parameter) {
+    final dart_ast.TypeName typeName = parameter.childEntities.firstWhere(
+        (dart_ast_syntactic_entity.SyntacticEntity e) =>
+            e is dart_ast.TypeName) as dart_ast.TypeName;
+    final String argTypeBaseName = typeName.name.name;
+    final bool isNullable = typeName.question != null;
+    final List<TypeDeclaration>? argTypeArguments =
+        typeAnnotationsToTypeArguments(typeName.typeArguments);
+    return NamedType(
+        type: TypeDeclaration(
+            baseName: argTypeBaseName,
+            isNullable: isNullable,
+            typeArguments: argTypeArguments),
+        name: parameter.identifier?.name ?? '',
```

Comment:
What would happen when it came time to output code if this were empty? Does the higher-level code handle it reasonably? If not, maybe this should throw instead of falling back to `''`.

Reply:
So, before this change the name was always '' so there are paths in the code that handle that and autogenerate names when there aren't defined one.  If you don't define a name you'll probably get a null pointer exception before you even get to that point from the `identifier?` expression.  I'll follow up to make sure people get a meaningful error if that is a problem instead of getting a hard to parse null exception.